### ISSUE TITLE
Add easy emoji syntax plugin

### ISSUE THREAD

**Issue by preaction** [preaction/Statocles#510]
A lot of websites and applications are standardizing on the `:<name>:` syntax for emoji and icons, like Github here.

* :+1: `:+1:`
* :100: `:100:`
* :cake: `:cake:`

It would be nice if we could put `:+1:` in our content and have it appear as :+1:. This would probably be best achieved with a `Statocles::Plugin::Emoji`. The plugin should scan the text content of every page and replace `:<name>:` with the matching icon. If an icon doesn't match, the text should remain the same and a warning printed out to warn the user that we didn't find the icon they wanted.

This plugin should be made part of the default set of plugins built by the site create command.

There needs to be a way to escape the emoji replacement. Likely by putting an escape backslash `\` in front of the first colon.

Since we already have font-awesome, and at least one OS fallback font has a wide array of Unicode emoji, we should be able to map a bunch of names to either font-awesome icons or Unicode emoji. It's up to you to decide what icons should be font-awesome, and what icons should be Unicode emoji.

It would be preferable if building the mapping of `<name>` to icon/emoji were automatic. If necessary, the proper Unicode specification could be downloaded, the names extracted and mapped to characters, and the appropriate font-awesome or Unicode character printed out. Otherwise, Perl has internal syntax for named Unicode characters (but that syntax is kind of obtuse).

The unicode characters will only work if we're writing out our pages using a Unicode encoding (like UTF-8). See #278 for work that needs to be done handling encodings better.

The work of short, memorable names for these icons/emoji may already exist on CPAN. If it doesn't this work may be spun out into its own CPAN distribution which we can then depend on.

**Comment by ferki** [comment]
[Text::SlackEmoji](https://metacpan.org/pod/Text::SlackEmoji) might be interesting here for the replacement (and for generating the emoji list from Unicode specs too, I believe).

**Comment by preaction** [comment]
Yes, that is perfect! So we basically just need to wrap this in a Statocles plugin.
